pkg/component/web.go

```diff
@@ -50,19 +50,20 @@ func (c *Component) RegisterReadinessCheck(name string, check healthcheck.Check)
 	c.healthHandler.AddReadinessCheck(name, check)
 }
 
-func (c *Component) serveHTTP(lis net.Listener) error {
+func (c *Component) serveWeb(lis net.Listener) error {
 	return http.Serve(lis, c)
 }
 
-func (c *Component) httpListenerConfigs() []endpoint {
-	return []endpoint{
-		{toNativeListener: Listener.TCP, address: c.config.HTTP.Listen, protocol: "HTTP"},
-		{toNativeListener: Listener.TLS, address: c.config.HTTP.ListenTLS, protocol: "HTTPS"},
+func (c *Component) webEndpoints() []Endpoint {
+	return []Endpoint{
+		NewTCPEndpoint(c.config.HTTP.Listen, "HTTP"),
```

Comment:
Bit confusing that on one hand we're changing HTTP to web, while here we're still using HTTP, but okay...